### ISSUE TITLE
Opportunities to improve the online documentation

### ISSUE THREAD

**Issue by kandersolar** [SunPower/pvfactors#143]
Here's are two things that would be nice to have in the docs, derived largely or entirely from conversation with @spaneja:

## Relevant class methods don't have their own docs page

For example: PVEngine has its own page ([link](https://sunpower.github.io/pvfactors/developer/generated/pvfactors.engine.PVEngine.html)), but its `fit()` method does not, making it hard to know what parameters the method takes (without referring to the tutorials, anyway).  I think it should be straightforward to get this working just with sphinx configuration changes.

## Irradiance component names and meanings are not documented

It's possible to extract not only total incident and absorbed irradiance but also various sub-components.  The docs have an example of this:

https://github.com/SunPower/pvfactors/blob/21d48655ce6acd3ab1955f98212cbdd3e3cc8351/docs/tutorials/Run_full_timeseries_simulations.ipynb?short_path=2c386fd#L232-L235

It would be nice for the available component names (e.g. `reflection`) to be listed somewhere, along with specific descriptions of what each name refers to (e.g. does `reflection` include ground-reflected irradiance, or just row-row reflections?).

**Comment by anomam** [comment]
yes the documentation is lacking in a lot of ways, sorry... I didn't have much time and help back then.
I see a lot of areas that can be improved even in the code, for instance these values `'reflection'`, `'isotropic'`, ... should clearly be enums...
